ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,242 @@
+                           Ikarus::Impl::VectorizeWithVoigt<strainlike>, Ikarus::Impl::MatricizeWithVoigt<strainlike>)
+
+  /**
+   * \brief Used to refister a general ResultType with potentially dynamic size without reserved memeroy
+   * rows and columns
+   * \param resultTypeName name of the ResultType
+   * \param rowsExpr expression for rows, e.g. `Eigen::Dynamic` or `worldDim`
+   * \param colsExpr expression for columns, e.g. `Eigen::Dynamic` or `worldDim`
+   */
+#define REGISTER_RESULTTYPE(resultTypeName, rowsExpr, colsExpr)                                \
+  REGISTER_RESULTTYPE_IMPL(resultTypeName, rowsExpr, colsExpr, Ikarus::Impl::VectorizeGeneric, \
+                           Ikarus::Impl::MatricizeGeneric)
+  /**
+   * \brief Used to refister a general ResultType with potentially dynamic size and defined maximum amount of
+   * rows and columns
+   * \param resultTypeName name of the ResultType
+   * \param rowsExpr expression for rows, e.g. `Eigen::Dynamic` or `worldDim`
+   * \param colsExpr expression for columns, e.g. `Eigen::Dynamic` or `worldDim`
+   * \param MaxRowsExpr expression for maximum number of columns
+   * \param MaxColsExprexpression for maximum number of rows
+   */
+#define REGISTER_RESERVED_RESULTTYPE(resultTypeName, rowsExpr, colsExpr, MaxRowsExpr, MaxColsExpr) \
+  REGISTER_RESULTTYPE_IMPL(resultTypeName, rowsExpr, colsExpr, MaxRowsExpr, MaxColsExpr,           \
+                           Ikarus::Impl::VectorizeGeneric, Ikarus::Impl::MatricizeGeneric)
+/**
+ * \brief Used to refister a general ResultType with potentially dynamic size, with reserved memory according to
+ * passed in rowsExpr and colsExpr
+ * \param resultTypeName name of the ResultType
+ * \param rowsExpr expression for rows, e.g. `Eigen::Dynamic` or `worldDim`
+ * \param colsExpr expression for columns, e.g. `Eigen::Dynamic` or `worldDim`
+ */
+#define REGISTER_SIMPLE_RESULTTYPE(resultTypeName, rowsExpr, colsExpr) \
+  REGISTER_RESERVED_RESULTTYPE(resultTypeName, rowsExpr, colsExpr, rowsExpr, colsExpr)
+
+  ///=============================
+  /// Registered ResultTypes
+
+  REGISTER_SIMPLE_SYMMETRIC_RESULTTYPE(linearStress, worldDim, worldDim, false);
+  REGISTER_SIMPLE_SYMMETRIC_RESULTTYPE(PK2Stress, worldDim, worldDim, false);
+  REGISTER_SIMPLE_SYMMETRIC_RESULTTYPE(cauchyStress, worldDim, worldDim, false);
+
+  REGISTER_SIMPLE_RESULTTYPE(director, worldDim, 1);
+  REGISTER_SIMPLE_RESULTTYPE(magnetization, worldDim, 1);
+  REGISTER_SIMPLE_RESULTTYPE(gradientNormOfMagnetization, 1, 1);
+  REGISTER_SIMPLE_RESULTTYPE(vectorPotential, worldDim, 1);
+  REGISTER_SIMPLE_RESULTTYPE(divergenceOfVectorPotential, 1, 1);
+
+  REGISTER_SIMPLE_RESULTTYPE(BField, worldDim, 1);
+  REGISTER_SIMPLE_RESULTTYPE(HField, worldDim, 1);
+
+  REGISTER_SIMPLE_RESULTTYPE(customType, Eigen::Dynamic, Eigen::Dynamic);
+} // namespace ResultType
+
+enum class ResultShape
+{
+  Vector,
+  Matrix,
+  Scalar
```

Comment:
I would remove scalar here. This is actually not relly supported anyway for the rest of the code, is it?